### PR TITLE
Fixed quote add/edit forms not validating input properly

### PR DESCRIPTION
**Pull request by ace-n**
Finally removed that stubborn dependencies.txt/requirements.txt file

### CODE REVIEW

File: liquid/intranet/quote/views.py [hunk 1]
```diff
@@ -129,10 +133,14 @@ def edit(request, quoteId = 1):
       
       # Convert <br />'s into newlines (\n - TODO?: this may cause issues for Windows users)
       quote_obj.quote_text = string.replace(quote_obj.quote_text, "<br />", "\n")
-
+
```
Comment (colegleason) [inline liquid/intranet/quote/views.py]: revert

File: liquid/intranet/quote/views.py [hunk 2]
```diff
@@ -129,10 +133,14 @@ def edit(request, quoteId = 1):
...
       quote_form = QuoteForm(instance=quote_obj)
-      quote_form.fields["quote_posters"].widget = forms.HiddenInput()
-       
-      # -- Handle quote editing --
-      return render_to_response('intranet/quote/edit.html',{"section":"intranet","page":'quote',"form":quote_form, "members":Member.objects.all(),"quoteMembers":quoteMembers,"quote_id":quoteId,"user":request.user},context_instance=RequestContext(request))  
+
+   quote_form.fields["quote_posters"].widget = forms.HiddenInput()
+   
+   # --- Return form (either a new one, or one with existing data if user has provided invalid data) ---
+   print author_usernames
```
Comment (colegleason) [inline liquid/intranet/quote/views.py]: remove

**Comment by colegleason** [comment]
Please fix the last two comments and I will merge this, assuming it is still needed.

**Comment by sskhandek** [comment]
@ace-n , can we close this asap?

File: liquid/intranet/quote/models.py [hunk 3]
```diff
@@ -56,11 +58,14 @@ def save(self, *args, **kwargs):
...
-      self.quote_posters = "," + html.escape(re.sub(",\s+", ",", self.quote_posters).strip(",")) + ","
-      posters = self.quote_posters.strip(",").split(",")
+      if len(self.quote_posters) != 0: # Posters were not recorded for some earlier quotes
+         self.quote_posters = "," + html.escape(re.sub(",\s+", ",", self.quote_posters).strip(",")) + ","
+         posters = self.quote_posters.strip(",").split(",")
+      else:
+         posters = dict() # Posters should contain no elements if self.quote_posters is blank
          
       # If the quote posters are different than the quote authors, create the quote poster HTML
-      posters_are_authors = len(posters) == len(authors);
```
Comment (ace-n) [inline liquid/intranet/quote/models.py]: [http://cdn.meme.am/instances/500x/64783469.jpg](url)

...Whoops, let me fix that.

**Comment by ace-n** [comment]
@sskhandek @colegleason Care to scan this?

Admittedly, we should probably to refactor this to use JSON sometime soon. It looks like there might be a security (specifically an XSS) vulnerability here in the quote-poster-HTML logic, though it is internal-only and probably better placed in a separate PR.

**Pull request closed without merging**